api/v1beta1/frrconfiguration_webhook.go

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var (
+	Logger        log.Logger
+	WebhookClient client.Reader
+	Validate      func(resources ...client.ObjectList) error
+	Namespace     string
+)
+
+func (frrConfig *FRRConfiguration) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(frrConfig).
+		Complete()
+}
+
+//+kubebuilder:webhook:verbs=create;update,path=/validate-frrk8s-metallb-io-v1beta1-frrconfiguration,mutating=false,failurePolicy=fail,groups=frrk8s.metallb.io,resources=frrconfigurations,versions=v1beta1,name=frrconfigurationsvalidationwebhook.metallb.io,sideEffects=None,admissionReviewVersions=v1
+
+var _ webhook.Validator = &FRRConfiguration{}
+
+type nodeAndConfigs struct {
+	name   string
+	labels map[string]string
+	cfgs   *FRRConfigurationList
+}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for FRRConfiguration.
+func (frrConfig *FRRConfiguration) ValidateCreate() error {
+	level.Debug(Logger).Log("webhook", "frrconfiguration", "action", "create", "name", frrConfig.Name, "namespace", frrConfig.Namespace)
+
+	if frrConfig.Namespace != Namespace {
```

Comment:
We currently allow the resource to live in multiple namespaces, and I am not sure we want to enforce it.
For metallb for example, it makes sense for it to create them in its own namespace.

Reply:
done